src/graph/network/serialize.h

```diff
@@ -0,0 +1,53 @@
+/*!
+ *  Copyright (c) 2019 by Contributors
+ * \file serialize.h
+ * \brief Serialization for DGL distributed training.
+ */
+#ifndef DGL_GRAPH_NETWORK_SERIALIZE_H_
+#define DGL_GRAPH_NETWORK_SERIALIZE_H_
+
+#include <dgl/sampler.h>
+#include <dgl/immutable_graph.h>
+
+namespace dgl {
+namespace network {
+
+/*!
+ * \brief Serialize sampled subgraph to binary data
+ * \param data pointer of data buffer
+ * \param csr subgraph csr
+ * \param node_mapping node mapping in NodeFlowIndex
+ * \param edge_mapping edge mapping in NodeFlowIndex
+ * \param layer_offsets layer offsets in NodeFlowIndex
+ * \param flow_offsets flow offsets in NodeFlowIndex
+ * \return the total size of the serialized binary data
+ */
+int64_t SerializeSampledSubgraph(char* data,
+                                 const ImmutableGraph::CSR::Ptr csr,
+                                 const IdArray& node_mapping,
+                                 const IdArray& edge_mapping,
+                                 const IdArray& layer_offsets,
+                                 const IdArray& flow_offsets);
```

Comment:
this interface is defined specifically for the current NodeFlow. It'll be extended after NodeFlow from random walker sampler is added. Can we make this interface a little more flexible?

Reply:
I discussed this with @BarclayII  and find that it is not very easy to change this interface.